clickhouse: 22.8.5.29 -> 23.3.1.2823

theduke opened the pull request:
###### Description of changes

Upgrade Clickhouse to v23.3.1.2823-lts

Closes #227435


###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

theduke committed:
clickhouse: 22.8.5.29 -> 23.3.1.2823

Upgrade clickhouse

Closes #227435

theduke commented:
cc @orivej

theduke commented:
Not sure if the two passthru test failures are something to be concerned about?

happysalada commented:
I've just tested building "23.3.2.37"
would you mind upgrading once more ?

You can also ping me for future clickhouse updates.

happysalada commented:
The ci failures look like building failures, since this builds I would say we can ignore those.

happysalada commented:
actually it looks like that next version is going to be a separate upgrade. I'm getting failures in the build. let me test this pullrequest, it looks like it's better to merge this first.

teh commented:
I'm interested in this as well. @theduke can I help in any way?

theduke commented:
I fixed the hash and added llvm15 and yasm dependencies. 

Getting quite a bit further now, but the build is failing with:

```
FAILED: base/glibc-compatibility/CMakeFiles/glibc-compatibility.dir/glibc-compatibility.c.o
/nix/store/669lrl7j7g4315xv247s8vm7flp0wwxb-clang-wrapper-15.0.7/bin/clang --target=x86_64-linux-gnu --sysroot=/build/source/cmake/linux/../../contrib/sysroot/linux-x>
/build/source/base/glibc-compatibility/glibc-compatibility.c:167:21: error: attribute declaration must precede definition [-Werror,-Wignored-attributes]
void __attribute__((__weak__)) explicit_bzero(void * buf, size_t len)
                    ^
/nix/store/bhhb91angi23wsr42apz1in1n9b9ilg3-glibc-2.37-8-dev/include/bits/string_fortified.h:70:8: note: previous definition is here
__NTH (explicit_bzero (void *__dest, size_t __len))
```

theduke commented:
A simple `  CMAKE_CXX_FLAGS = "-wignored-attributes";` doesn't do the trick.

PS: happy for anyone to take over, I don't have much experience with building cmake/C++ projects.

teh commented:
maybe try `llvmPackages_15.stdenv.mkDerivation`?

yu-re-ka commented:
`env.NIX_CFLAGS_COMPILE = "-Wno-ignored-attributes";` brings it one step further

yu-re-ka commented:
top-level.nix also contains references to llvmPackages_14 that are passed into the clickhouse callPackage
